sycl/include/sycl/detail/string.hpp

```diff
@@ -0,0 +1,82 @@
+//==----------------- string.hpp - SYCL standard header file ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstring>
+
+#pragma once
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+
+// This class and detail::string_view class are intended to support
+// different ABIs between libsycl and the user program.
+// This class is not inteded to replace std::string for general purpose usage.
+class string {
+  char *str =
+      nullptr; // set from libsycl to return a std::string to a user program
+
+public:
+  string() = default;
```

Comment:
Should this constructor allocate a single `\0`-byte? Otherwise the copy constructor and the `==`-operators will potentially pass `nullptr` to `strlen` respectively `strcmp`.

Reply:
Thanks.